src/pages/index.js

```diff
@@ -3,120 +3,247 @@ import "../pages/index.css";
+    },
+    (cardId, isLiked, cardElement) => {
+      handleLikeClick(cardId, isLiked, cardElement, card);
+    }
   );
   return card.getView();
 }
 const cardsSection = new Section(
   {
-    items: initialCards,
     renderer: (data) => {
       const cardElement = createCard(data);
       cardsSection.addItem(cardElement);
     },
   },
   ".cards__list"
 );
-cardsSection.renderItems();
 
-const addCardWithForm = new PopupWithForm(
-  {
-    popupSelector: "#add-card-modal",
-  },
-  handleAddCardSubmit
-);
+api
+  .getInitialCards()
+  .then((result) => {
+    cardsSection.renderItems(result);
+  })
+  .catch((err) => console.log("Error loading initial cards:", err));
 
-addCardWithForm.setEventListeners();
-
-addNewCardBtn.addEventListener("click", () => addCardWithForm.open());
-
-//user info
+/*---------User info---------*/
 
 const userInfo = new UserInfo({
   profileTitle: ".profile__title",
   profileDescription: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+api
+  .getUserInfo()
+  .then((result) => {
+    if (userInfo) {
+      userInfo.setUserInfo({
+        title: result.name,
+        description: result.about,
+      });
+      userInfo.setUserAvatar(result.avatar);
+    }
+  })
+  .catch((err) => console.log("Error loading user info:", err));
+
 function fillUserData() {
   const userData = userInfo.getUserInfo();
   profileEditTitle.value = userData.title;
   profileEditDescription.value = userData.description.trim();
 }
 
+/*----------Popups-----*/
+
+//Edit Profile Popup
 const editModalWithForm = new PopupWithForm(
-  {
-    popupSelector: "#profile-edit-modal",
-  },
+  "#profile-edit-modal",
   handleProfileEditSubmit
 );
 
 editModalWithForm.setEventListeners();
 
-function handleImageClick(imageName, imageLink) {
-  imagePopup.open({ name: imageName, link: imageLink });
-}
+//Add new Card Popup
+const addCardWithForm = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardSubmit
+);
 
-const imagePopup = new PopupWithImage({
-  popupSelector: "#modal-image-preview",
-});
+addCardWithForm.setEventListeners();
+
+addNewCardBtn.addEventListener("click", () => addCardWithForm.open());
+
+//Image preview Popup
+const imagePopup = new PopupWithImage("#modal-image-preview");
 
 imagePopup.setEventListeners();
 
+//Delete card Confirm Popup
+const deleteCardPopup = new PopupDeleteCard({
+  popupSelector: "#delete-confirm-modal",
+});
+deleteCardPopup.setEventListeners();
+
+//Adit Avater Popup
+const avatarModal = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleEditAvatarFormSubmit
+);
+avatarModal.setEventListeners();
+
+profileAvatarButton.addEventListener("click", () => {
+  avatarModal.open();
+});
+
+function handleImageClick(imageName, imageLink) {
+  imagePopup.open({ name: imageName, link: imageLink });
+}
+
 //handle functions
 
 function handleProfileEditSubmit(value) {
-  userInfo.setUserInfo(value);
-  editModalWithForm.close();
+  editModalWithForm.setLoadingState(false);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```